R/browser.R

```diff
@@ -0,0 +1,124 @@
+##' Control browser-based of dust models.  This help page documents
+##' three funtions that can be used to control if and how the browser
+##' is enabled.  You can't enter the debugger from any of these
+##' functions; it is only enabled if present in your C++ code (or if
+##' using `odin2` if you have enabled it).
+##'
+##' dust2 includes an extremely simple debugging system, and if you
+##' are reading this message, there's a good chance you are inside it.
+##' It is built on top of R's [browser()] and so all the usual tips,
+##' tricks and issues for working with this apply.  We recommend
+##' setting the R option `browserNLdisabled = TRUE` to avoid surprises
+##' from presssing `<enter>`.
+##'
+##' * You can press `n` or `c` to proceed to the next enabled iteration
+##' * You can press `Q` to quit the browser (this will end up as an
+##'   error by the time you have control back)
+##'
+##' These commands are established by `browser`, and can't be
+##' disabled.  This means that if you have an variable called `n` you
+##' will need to work with it as `(n)` (i.e. in parentheses).  This
+##' applies to all of browser's command variables (`c`, `f`, `n`, `s`,
+##' `r` and `Q`); please see [browser] for more information.
+##'
+##' By the time the environment has been created, some variables from
+##' your model will have been copied into the environment; you can see
+##' these by running `ls()` and write expressions involving these
+##' objects.  Changes that you make in R are *not* (currently)
+##' propagated back into the running system.
```

Comment:
Does the "currently" imply you're thinking of an option to copy the debug environment back to the running one? Interesting...

Reply:
it does mean exactly that! not actually that hard to do